src/clients/createClient.ts

```diff
@@ -127,23 +131,25 @@ export type Client<
   account extends Account | undefined = Account | undefined,
   rpcSchema extends RpcSchema | undefined = undefined,
   extended extends Extended | undefined = Extended | undefined,
+  mode extends TestClientMode | undefined = undefined,
 > = Client_Base<transport, chain, account, rpcSchema> &
   (extended extends Extended ? extended : unknown) & {
     extend: <
       const client extends Extended &
         ExactPartial<ExtendableProtectedActions<transport, chain, account>>,
     >(
       fn: (
-        client: Client<transport, chain, account, rpcSchema, extended>,
+        client: Client<transport, chain, account, rpcSchema, extended, mode>,
       ) => client,
     ) => Client<
       transport,
       chain,
       account,
       rpcSchema,
-      Prettify<client> & (extended extends Extended ? extended : unknown)
+      Prettify<client> & (extended extends Extended ? extended : unknown),
+      mode
     >
-  }
+  } & (mode extends TestClientMode ? { mode: mode } : {})
```

Comment:
```suggestion
  } & (mode extends TestClientMode ? { mode: mode } : {mode?: never})
```
Maybe should be this?